pkg/providers/amifamily/suite_test.go

```diff
@@ -196,6 +197,56 @@ var _ = Describe("AMIProvider", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(amis).To(HaveLen(0))
 	})
+	It("should not cause data races when calling Get() simultaneously", func() {
+		nodeClass.Spec.AMISelectorTerms = []v1beta1.AMISelectorTerm{
+			{
+				ID: "amd64-ami-id",
+			},
+			{
+				ID: "arm64-ami-id",
+			},
+		}
+		wg := sync.WaitGroup{}
+		for i := 0; i < 100000; i++ {
```

Comment:
is 100k really necessary here? I'm not really sure how this impacts performance of our tests, so it might be fine.

Reply:
Can do 10,000. I was able to repro it at that scale. This was still only taking ~4-6s so 100,000 is also fine.